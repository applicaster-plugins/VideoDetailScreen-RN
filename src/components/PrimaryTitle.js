import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import Text from '@applicaster/london-rn-components/src/SafeText';
import { styles } from './PrimaryTitle.styles';

class PrimaryTitle extends PureComponent {
  constructor(props) {
    super(props);
    // deferred for layout data to be ready
    this.styles = styles();
  }

  render() {
    const { children } = this.props;
    return (
      <View style={this.styles.container}>
        <Text style={this.styles.text}>{children}</Text>
      </View>
    );
  }
}

PrimaryTitle.propTypes = {
  children: PropTypes.string
};

export default PrimaryTitle;
