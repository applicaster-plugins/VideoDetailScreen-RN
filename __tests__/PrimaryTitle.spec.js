import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import PrimaryTitle from '../src/components/PrimaryTitle';

jest.mock('../src/utils/zappConfig');

it('PrimaryTitle renders correctly', () => {
  const tree = renderer
    .create(<PrimaryTitle>Test Title</PrimaryTitle>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
