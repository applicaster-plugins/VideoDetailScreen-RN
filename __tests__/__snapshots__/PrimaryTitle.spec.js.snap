// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryTitle renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "justifyContent": "flex-start",
      "marginVertical": 4,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "rgb(239, 239, 239)",
        "fontFamily": "Exo2-SemiBold",
        "fontSize": 24,
        "letterSpacing": 0,
        "lineHeight": 28,
        "textAlign": "left",
      }
    }
  >
    Test Title
  </Text>
</View>
`;
